components/segmenter/src/provider/mod.rs

```diff
@@ -200,3 +201,68 @@ pub(crate) struct UCharDictionaryBreakDataV1Marker;
 impl DataMarker for UCharDictionaryBreakDataV1Marker {
     type Yokeable = UCharDictionaryBreakDataV1<'static>;
 }
+
+#[derive(Clone, Copy, PartialEq, Debug)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(databake::Bake),
+    databake(path = icu_segmenter::provider),
+)]
+/// Break state
+pub enum BreakState {
+    /// Break
+    Break,
+    /// Keep rule
+    Keep,
+    /// Non-matching rule
+    NoMatch,
+    /// We have to look ahead one more character.
+    Intermediate(u8),
+    /// Index of a property.
```

Comment:
I think this can be the index of any state, not just one that corresponds to a property.